src/CoreWCF.Http/tests/Helpers/ServiceHelper.cs

```diff
@@ -126,6 +128,7 @@ public static IWebHostBuilder CreateWebHostBuilder<TStartup>(ITestOutputHelper o
             CreateBaseWebHostBuilder<TStartup>(outputHelper, callerMethodName)
             .UseKestrel(options =>
             {
+                options.AllowSynchronousIO = true;
```

Comment:
This enables synchronous IO for all tests. We don't want this always on. If a change is made which regresses the other scenarios and starts writing out synchronously, we might miss it and ship a pretty nasty bug. You should be able to configure kestrel in the test which calls this method. In the calling test you can do something like this:
```c#
var hostBuilder = ServiceHelper.CreateWebHostBuilder(_output, startupType);
hostBuilder.ConfigureServices(services =>
{
   services.Configure<KestrelServerOptions>(options =>
   {
       options.AllowSynchronousIO = true;
   });
});
IWebHost host = hostBuilder.Build();
```

Reply:
You could make a helper extension method in ServiceHelper for this to make it a one lines. Like this:
ServiceHelper.cs
```c#
public static IWebHostBuilder AllowSynchronousIO(this IWebHostBuilder webHostBuilder)
{
    webHostBuilder.ConfigureServices(services =>
    {
       services.Configure<KestrelServerOptions>(options =>
       {
           options.AllowSynchronousIO = true;
       });
    });
    return webHostBuilder;
}
```
In unit test method:
```c#
IWebHost host = ServiceHelper.CreateWebHostBuilder<Startup>(_output)
                .AllowSynchronousIO()
                .Build();
```